static/app/views/issueDetails/groupActivity.tsx

```diff
@@ -1,219 +1,111 @@
-   * This can be abstracted a bit if we create more objects that can have activities
-   */
-  handleNoteCreate = async note => {
-    const {api, group, organization} = this.props;
-
-    this.setState({
-      createBusy: true,
-    });
-
-    addLoadingMessage(t('Posting comment\u{2026}'));
-
-    try {
-      await createNote(api, organization.slug, group, note);
-
-      this.setState({
-        createBusy: false,
-
-        // This is used as a `key` to Note Input so that after successful post
-        // we reset the value of the input
-        inputId: uniqueId(),
-      });
-      clearIndicators();
-    } catch (error) {
-      this.setState({
-        createBusy: false,
-        error: true,
-        errorJSON: error.responseJSON || DEFAULT_ERROR_JSON,
-      });
+  const addMutationOptions: MutateOptions<TData, TError, TVariables, TContext> = {
+    onError: () => {
       addErrorMessage(t('Unable to post comment'));
-    }
+    },
+    onSuccess: data => {
+      GroupStore.addActivity(group.id, data);
+      addSuccessMessage(t('Comment posted'));
+    },
   };
 
-  handleNoteUpdate = async (note, {noteId, text: oldText}) => {
-    const {api, group, organization} = this.props;
-
-    addLoadingMessage(t('Updating comment\u{2026}'));
-
-    try {
-      await updateNote(api, organization.slug, group, note, noteId, oldText);
-      clearIndicators();
-    } catch (error) {
-      this.setState({
-        error: true,
-        errorJSON: error.responseJSON || DEFAULT_ERROR_JSON,
-      });
+  const updateMutationOptions: MutateOptions<TData, TError, TVariables, TContext> = {
+    onError: () => {
       addErrorMessage(t('Unable to update comment'));
-    }
+    },
+    onSuccess: data => {
+      const d = data as GroupActivityNote;
+      GroupStore.updateActivity(group.id, data.id, {text: d.data.text});
+      addSuccessMessage(t('Comment updated'));
+    },
   };
 
-  render() {
-    const {group, organization} = this.props;
-    const {activity: activities, count, id: groupId} = group;
-    const groupCount = Number(count);
-    const mostRecentActivity = getGroupMostRecentActivity(activities);
-    const reprocessingStatus = getGroupReprocessingStatus(group, mostRecentActivity);
-
-    const me = ConfigStore.get('user');
-    const projectSlugs = group && group.project ? [group.project.slug] : [];
-    const noteProps = {
-      minHeight: 140,
-      group,
-      projectSlugs,
-      placeholder: t(
-        'Add details or updates to this event. \nTag users with @, or teams with #'
-      ),
-    };
-
-    return (
-      <Fragment>
-        {(reprocessingStatus === ReprocessingStatus.REPROCESSED_AND_HASNT_EVENT ||
-          reprocessingStatus === ReprocessingStatus.REPROCESSED_AND_HAS_EVENT) && (
-          <ReprocessedBox
-            reprocessActivity={mostRecentActivity as GroupActivityReprocess}
-            groupCount={groupCount}
-            orgSlug={organization.slug}
-            groupId={groupId}
+  return (
```

Comment:
```suggestion
```suggestion
  const deleteMutationOptions = {
    onError: () => {
      addErrorMessage(t('An error occurred while removing the comment.'));
    },
    onSuccess: () => {
      addSuccessMessage(t('Comment removed'));
    },
  };
  
  const deleteCommentCallback = useCallback(() => {
                    const restore = group.activity.find(
                      activity => activity.id === item.id
                    );
                    const index = GroupStore.removeActivity(group.id, item.id);

                    if (index === -1 || restore === undefined) {
                      addErrorMessage(t('Failed to delete comment'));
                      return;
                    }
                    mutators.onDelete(...);
  }, [...]);

```


  return (
```